src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -201,14 +202,17 @@ private function parseDefinition($id, $service, $file)
         }
 
         if (isset($service['factory_class'])) {
+            trigger_error(sprintf('The "factory_class" option in file "%s" is deprecated since version 2.6 and will be removed in 3.0. Use Definition::setFactory() instead.', $file), E_USER_DEPRECATED);
```

Comment:
As we now it comes from a YAML file, the hint is not the right one. We should tell the dev how to configure it in a YAML context.